index.d.ts

```diff
@@ -0,0 +1,81 @@
+export interface Action {
+  type: string;
+}
+
+
+/* reducers */
+
+export type Reducer<S> = <A extends Action>(state: S, action: A) => S;
+
+export function combineReducers<S>(reducers: {[key: string]: Reducer<any>}): Reducer<S>;
+
+
+/* store */
+
+export interface Dispatch {
+  <A>(action: A): A;
+  <A, B>(action: A): B;
+}
+
+export interface Store<S> {
+  dispatch: Dispatch;
+  getState(): S;
+  subscribe(listener: () => void): () => void;
```

Comment:
One more lil suggestion here - lets create type/interface for function returned from subscribe so user can declare variable which will hold reference to it in future:

```typescript
// index.d.ts
export interface Unsubscribe extends Function {
    (): void;
}

export interface Store<S> {
    …
    subscribe(listener: () => void): Unsubscribe;
    …
}

// real example from my code:
@Component({selector: 'my-app', templateUrl: '/src/app.html'})
export class App implements OnInit, OnDestroy {
    constructor(@Inject('AppStore') private appStore: Store<IAppState>) {}
    …
    
    // I can declare it with proper type;
    private unsubscribe: Unsubscribe;
    
    // and assign value later
    public ngOnInit(): void {
        this.unsubscribe = this.appStore.subscribe(() => …doSomethingWith(this.appStore.getState())… );
    }

    public ngOnDestroy(): void {
        this.unsubscribe();
    }

}
```